ToolBox/TickAggregatingConsolidator.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Collections.Generic;
+using QuantConnect.Data;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.ToolBox
+{
+    /// <summary>
+    /// Class that uses consolidators to aggregate tick data data
+    /// </summary>
+    public abstract class TickAggregatingConsolidator
```

Comment:
*Aggregating* consolidator almost seems redundant. I'm not the best with names but to me this type is able to perform consolidation *and* keep track of the history of what's been consolidated.